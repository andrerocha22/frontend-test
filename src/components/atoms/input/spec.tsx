import React from 'react';
import { shallow, mount } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import { InputCurrency } from './index';

const mockCallBack = jest.fn();

const defaultProps = {
  icon: 'iconTest',
};

const setup = (props = {}) => {
  const setupProps = { ...props };
  return shallow(
    <InputCurrency onChangeCallback={mockCallBack} {...setupProps} />
  );
};

describe('Test currency input', () => {
  test('renders without error', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-currency-input');
    expect(component.length).toBe(1);
  });

  test('renders without error with icon props', () => {
    const wrapper = setup(defaultProps);
    const component = findByTestAttr(
      wrapper,
      'component-currency-input-with-icon'
    );
    expect(component.length).toBe(1);
  });

  test('not renders icon without icon props', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-currency-input-icon');
    expect(component.length).toBe(0);
  });

  test('currency input snapshot test', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-currency-input');
    expect(component).toMatchSnapshot();
  });

  test('test callback', () => {
    const wrapper = mount(<InputCurrency onChangeCallback={mockCallBack} />);
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: '50000' } });
    expect(mockCallBack).toHaveBeenCalledTimes(1);
  });
});
