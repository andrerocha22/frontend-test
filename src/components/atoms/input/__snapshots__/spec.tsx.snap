// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test currency input currency input snapshot test 1`] = `
<styled.div
  data-test="component-currency-input"
>
  <CurrencyInput
    allowNegative={false}
    autoFocus={true}
    className="inputAmount"
    decimalSeparator="."
    inputType="text"
    maxLength="12"
    onChange={[Function]}
    onChangeEvent={[Function]}
    precision="2"
    prefix=""
    selectAllOnFocus={true}
    style={
      Object {
        "background": "#ffffff",
        "border": "none",
        "display": "flex",
        "fontSize": "2rem",
        "fontWeight": "600",
        "lineHeight": "2.4rem",
        "padding": "1.6rem",
        "width": "100%",
      }
    }
    suffix=""
    thousandSeparator=","
    value="0"
  />
</styled.div>
`;
